aws/vpc.go

```diff
@@ -0,0 +1,227 @@
+			if err != nil {
+				logrus.Error(err)
+			}
+		}
+	} else {
+		logrus.Infof("      no subnets found for %s", vpc.VpcID)
+	}
+}
+
+func (vpc Vpc) deleteRouteTables() {
+	routeTables, err := vpc.client.DescribeRouteTables(&ec2.DescribeRouteTablesInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{aws.String(vpc.VpcID)},
+			},
+		},
+	})
+	if err != nil {
+		logrus.Error(err)
+	}
+
+	if len(routeTables.RouteTables) > 0 {
+		logrus.Infof("      deleting route tables for %s", vpc.VpcID)
+
+		for _, routeTable := range routeTables.RouteTables {
+			if len(routeTable.Associations) > 0 && *routeTable.Associations[0].Main {
+				continue
+			}
+
+			_, err := vpc.client.DeleteRouteTable(&ec2.DeleteRouteTableInput{RouteTableId: routeTable.RouteTableId})
+			if err != nil {
+				logrus.Error(err)
+			}
+		}
+	} else {
+		logrus.Infof("      no route tables found for %s", vpc.VpcID)
+	}
+}
+
+func (vpc Vpc) deleteNACLs() {
+	nacls, err := vpc.client.DescribeNetworkAcls(&ec2.DescribeNetworkAclsInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{aws.String(vpc.VpcID)},
+			},
+		},
+	})
+	if err != nil {
+		logrus.Error(err)
+	}
+
+	if len(nacls.NetworkAcls) > 0 {
+		logrus.Infof("      deleting nacls for %s", vpc.VpcID)
+		for _, nacl := range nacls.NetworkAcls {
+			if !*nacl.IsDefault {
+				_, err := vpc.client.DeleteNetworkAcl(&ec2.DeleteNetworkAclInput{NetworkAclId: nacl.NetworkAclId})
+				if err != nil {
+					logrus.Error(err)
+				}
+			}
+		}
+	} else {
+		logrus.Infof("      no subnets found for %s", vpc.VpcID)
+	}
+}
+
+func (vpc Vpc) deleteSecurityGroups() {
+	sgs, err := vpc.client.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{aws.String(vpc.VpcID)},
+			},
+		},
+	})
+	if err != nil {
+		logrus.Error(err)
+	}
+
+	if len(sgs.SecurityGroups) > 0 {
+		logrus.Infof("      deleting security groups for %s", vpc.VpcID)
+		for _, sg := range sgs.SecurityGroups {
+			if *sg.GroupName != "default" {
+				_, err := vpc.client.DeleteSecurityGroup(&ec2.DeleteSecurityGroupInput{GroupId: sg.GroupId})
+				if err != nil {
+					logrus.Error(err)
+				}
+			}
+		}
+	} else {
+		logrus.Infof("      no security groups found for %s", vpc.VpcID)
+	}
+}
+
+func (vpc Vpc) deleteVpc() {
+	logrus.Infof("      deleting %s", vpc.VpcID)
+	_, err := vpc.client.DeleteVpc(&ec2.DeleteVpcInput{VpcId: &vpc.VpcID})
+	if err != nil {
+		logrus.Error(err)
+	}
+}
+
+func (vpc Vpc) delete() {
+	vpc.deleteInternetGateways()
+	vpc.deleteSubnets()
+	vpc.deleteRouteTables()
+	vpc.deleteNACLs()
+	vpc.deleteSecurityGroups()
+	vpc.deleteVpc()
+}
+
+// DeleteDefaultVPCs deletes all of the default VPCs in all regions of an account
+func DeleteDefaultVPCs(region string, role string, deleteFlag bool) error {
+	enabledRegions := GetEnabledRegions(region, role)
```

Comment:
if it does it in all regions, why do we provide `region`?

Reply:
In order to get the list of enabled regions, we have to provide a "seed" region to call the API to get a list of regions.